gatsby/src/pages/search.tsx

```diff
@@ -1,14 +1,278 @@
 import { SEO } from "../components/SEO/SEO";
 
-const SearchPage: React.FC = () => {
+type SearchPageProps = PageRendererProps;
+
+interface SearchResults {
+	failed: boolean;
+	resultCount: number;
+	firstResult: number;
+	lastResult: number;
+	finalSearchText: string;
+	finalSearchTextNoStopWords: string;
+	pageSize: number;
+	originalSearch: {
+		searchText: string;
+	};
+	pagerLinks: {
+		previous: {
+			fullUrl: string;
+		} | null;
+		next: {
+			fullUrl: string;
+		} | null;
+		pages: {
+			title: string;
+			url: {
+				fullUrl: string;
+			};
+			isCurrent: boolean;
+		}[];
+	};
+	documents: Document[];
+}
+
+type Document = {
+	id: string;
+	metaDescription: string;
+	contentId: string;
+	pathAndQuery: string;
+	sourceUrl: string;
+	teaser: string;
+	title: string;
+	url: string;
+};
+
+function titleString(text: string, page: number): string {
+	return text
+		? page > 1
+			? `Page ${page} | ${text} | `
+			: `${text} | `
+		: page > 1
+		? `Page ${page} | `
+		: "";
+}
+
+const SearchPage: React.FC<SearchPageProps> = ({
+	location,
+}: SearchPageProps) => {
+	const [data, setData] = useState<SearchResults | null>(null);
+	const [error, setError] = useState<Error>();
+
+	useEffect(() => {
+		setData(null);
+		fetch("/api/search" + window.location.search)
+			.then(data => data.json())
+			.then(results => setData(results as SearchResults))
+			.catch(e => setError(e));
+	}, [location.search]);
+
 	return (
 		<Layout>
-			<SEO title="Search" />
 			<h1>Search</h1>
+			{!error && !data && <div>Loading</div>}
+			{data && data.failed && <div>Failed to load results</div>}
+			{data && !data.failed && <Results {...data} />}
+			{error?.message && (
+				<Alert type="error">
+					<pre>
+						{error.name}
+						<br />
+						{error.message}
+						<br />
+						{error.stack}
+					</pre>
+				</Alert>
+			)}
 		</Layout>
 	);
 };
 
+const Results: React.FC<SearchResults> = ({
+	firstResult,
+	resultCount,
+	finalSearchText,
+	originalSearch,
+	documents,
+	pageSize,
+	pagerLinks: { next, previous },
+}) => {
+	const currentPage = Math.ceil(firstResult / pageSize);
+	const totalPages = Math.ceil(resultCount / pageSize);
+	const paginationProps = {
+		nextPageLink: next?.fullUrl
+			? { destination: "/" + next?.fullUrl, elementType: Link }
+			: undefined,
+		previousPageLink: previous?.fullUrl
+			? { destination: "/" + previous?.fullUrl, elementType: Link }
+			: undefined,
+		currentPage,
+		totalPages,
+	};
+
+	return (
+		<>
+			<SEO
+				title={`${titleString(finalSearchText, currentPage)}Search results`}
+				noIndex={true}
+			/>
+			<ResultSummary
+				resultCount={resultCount}
+				finalSearchText={finalSearchText}
+				originalSearch={originalSearch?.searchText}
+			/>
+			{documents.length > 0 && <ResultsList documents={documents} />}
+			{resultCount > pageSize && <SimplePagination {...paginationProps} />}
+		</>
+	);
+};
+
+interface ResultsSummary {
+	resultCount: number;
+	finalSearchText: string;
+	originalSearch: string;
```

Comment:
Can we call this prop `originalSearchText` instead to avoid any confusion with `originalSearch` on the returned JSON?

Reply:
good spot